OSCompatibilityLayer.h

```diff
@@ -1,37 +1,35 @@
 #ifndef OS_COMPATABILITY_LAYER_H
```

Comment:
Tests aren't strictly necessary, as they haven't existed before, but I wouldn't object if you added them.

I've recently added a few tests to Vic2ToHoI4 that do filesystem things and that can serve as a guide. You'll need to include test data and be aware of where it gets placed and where the tests are running, but it can be done. Visual Studio is actually pretty good about placing files if you include them in the project. I've been doing that in Vic2ToHoI4 for the Windows build for everything but blankMod. Sadly I don't know how to do it in CMake for the Linux build.

Reply:
Yeah, I'm not so hot about doing OS-level tests. 

These functions, as they are now, in OSCommonLayer.cpp, they are wrappers around std::filesystem, with added logging, sometimes not even that (DoesFileExist made sense before, but now it's just a fs::exists()). 

Also they are used commonly throughout all converters so any deviance would immediately be spotted as conversions would fail.

If there's a specific function relating to a specific task, like renaming something to something else, depending on user input, then sure, we can test that in the scope of that converter function, but for global OS-level functions, testing them seems redundant - they either work or they don't and if they don't converter won't run at all, ever.